src/test/java/test/tesng1046/TestCustomNamesForTests.java

```diff
@@ -0,0 +1,33 @@
+package test.tesng1046;
+
+import org.testng.Assert;
+import org.testng.ITestNGListener;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.collections.Sets;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+import test.SimpleBaseTest;
+
+import java.util.Set;
+
+public class TestCustomNamesForTests extends SimpleBaseTest {
+    @Test
+    public void testCustomTestNames() {
+        XmlSuite xmlSuite = createXmlSuite("Suite");
```

Comment:
I think we have a `createXmlSuite("Suite", "Test", TestClassSample.class)` method that limit boilerplate code ;)